group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -471,9 +472,48 @@ public CompletableFuture<OffsetCommitResponseData> commitOffsets(
             return FutureUtils.failedFuture(Errors.COORDINATOR_NOT_AVAILABLE.exception());
         }
 
-        return FutureUtils.failedFuture(Errors.UNSUPPORTED_VERSION.exception(
-            "This API is not implemented yet."
-        ));
+        if (request.groupId() == null) {
+            return CompletableFuture.completedFuture(OffsetCommitRequest.getErrorResponse(
+                request,
+                Errors.INVALID_GROUP_ID
+            ));
+        }
+
+        return runtime.scheduleWriteOperation(
+            "commit-offset",
+            topicPartitionFor(request.groupId()),
+            coordinator -> coordinator.commitOffset(context, request)
+        ).exceptionally(exception -> {
```

Comment:
This is pulled from here, right? https://github.com/apache/kafka/blob/1656591d0b339c385d0ba1f938fc94b52e29965d/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L437

Reply:
Correct. Note that `NOT_ENOUGH_REPLICAS_AFTER_APPEND` is not in the new code because this cannot happen.

Reply:
The mapping is different, right?

Reply:
ah i see. i thought they were the same

Reply:
Yeah, there are always slightly different. This is why I prefer to be explicit in the callback here.